src/core/DY.Core.Trace.Base.fst

```diff
@@ -820,9 +836,12 @@ let rec trace_search_last tr p =
 /// to decide if a given entry is the one we are looking for, and so we use this
 /// label-ignoring equivalence. This is exactly == for non-RandGen trace entries,
 /// but treats two RandGen entries with the same non-label content as equivalent.
-let trace_entry_equiv (#label_t:Type) (e1 e2:trace_entry_ label_t)
-  : bool
-  = (fmap_trace_entry forget_label e1) = (fmap_trace_entry forget_label e2)
+val trace_entry_equiv:
+  #label_t:Type ->
+  e1:trace_entry_ label_t -> e2:trace_entry_ label_t ->
```

Comment:
Nit: for non-dependent functions (like this one, and most non-Lemmas), we don't need to name most non-type arguments, so this could be 
```suggestion
  trace_entry_ label_t -> trace_entry_ label_t ->
```